jgrapht-core/src/main/java/org/jgrapht/alg/cycle/Cycle.java

```diff
@@ -0,0 +1,89 @@
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.WeightedGraph;
+import org.jgrapht.graph.GraphWalk;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A common abstract class for classes implementing algorithms for finding a cycle base of a graph.
+ *
+ * @param <V> the vertex type.
+ * @param <E> the edge type.
+ *
+ * @author Nikita Zhuchkov
+ * @author Stanislav Nikitin
+ * @author Svetlana Blinova
+ */
+
+public abstract class Cycle<V, E> {
+
+    protected Graph<V, E> graph;
+
+    /**
+     * Returns the graph on which the cycle base search algorithm is executed by this object.
+     *
+     * @return The graph.
+     */
+    abstract Graph<V, E> getGraph();
+
+    /**
+     * Sets the graph on which the cycle base search algorithm is executed by this object.
+     *
+     * @param graph the graph.
+     *
+     * @throws IllegalArgumentException if the argument is <code>null</code>.
+     */
+    abstract void setGraph(Graph<V, E> graph);
+
+
+    /**
+     * Finds a cycle base of the graph.<br>
+     * Note that the full algorithm is executed on every call since the graph may have changed
+     * between calls.
+     *
+     * @return A list of cycles constituting a cycle base for the graph. Possibly empty but never
+     *         <code>null</code>.
+     *
+     * @throws IllegalArgumentException if the current graph is null.
+     */
+    abstract List<List<V>> findCycles();
+
+    /**
+     * * Finds a cycle base of the graph.<br>
+     * Note that the full algorithm is executed on every call since the graph may have changed
+     * between calls.
+     *
+     * @return A list of paths which are cycles. Based on <code> List<List<V>> findCycles() </code>.
+     * @throws IllegalArgumentException
+     */
+    List<GraphPath<V, E>> findCyclePath() throws IllegalArgumentException {
```

Comment:
The implementation is already present in class `GraphWalk`. Simply call the method which returns a list of vertices and properly instantiate a `GraphWalk` to return to the users.